### PR TITLE
feat: t8 - display new tcp listening connections in the range 10000-10100

### PR DESCRIPTION
**Pull request by krasninja**
## Task

When you launch the script it should initially list all TCP connections in the port range 10000-10100 New found connections should be displayed in the console as soon as they're identified.

- File: `./chapter2/tests/t8/chapter2-l4-t8-get-tcp-listening-connections.sh`
- Task: https://github.com/saritasa-nest/saritasa-devops-camp/blob/main/chapter2%20-%20bash%20basics/l4/sockets.md#t8---display-new-tcp-listening-connections-in-the-range-10000-10100

## Usage

The `t8` directory contains sample Nginx configurations that can be used for testing.

```
$ nginx -p . -c nginx-10000.conf &
$ nginx -p . -c nginx-10001.conf &
```

Run the script in a separate terminal. The information about two connections should appear.

```
$ nginx -p . -c nginx-10002.conf &
```

The new connection should immediately appear.

```
killall nginx
```

### CODE REVIEW

File: chapter2/tests/t8/chapter2-l4-t8-get-tcp-listening-connections.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+# Output information about connections.
+# Globals:
+#   FILTER_EXPR
+#   total_ss_output
+# Arguments:
+#   None
+#######################################
+function get_connections {
+  # Compare last tracked connections and current ones. We want to display only new connections.
+  local ss_output=$(ss -ltn "${FILTER_EXPR}" --no-header)
```
Comment (kseniyashaydurova) [inline chapter2/tests/t8/chapter2-l4-t8-get-tcp-listening-connections.sh]:
```suggestion
  local ss_output=$(ss -ltnH "${FILTER_EXPR}")
```

File: chapter2/tests/t8/chapter2-l4-t8-get-tcp-listening-connections.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  fi
+
+  total_ss_output="${ss_output}"
+}
+
+while true; do
+  get_connections
+  sleep 2
+done
+
```
Comment (kseniyashaydurova) [inline chapter2/tests/t8/chapter2-l4-t8-get-tcp-listening-connections.sh]: Too many empty lines

File: chapter2/tests/t8/chapter2-l4-t8-get-tcp-listening-connections.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    fi
+    echo "${diff}" | awk '{ printf "%-20s %-20s\n", $4, $5 }'
+  fi
+
+  total_ss_output="${ss_output}"
+}
+
+while true; do
+  get_connections
+  sleep 2
```
Comment (kseniyashaydurova) [inline chapter2/tests/t8/chapter2-l4-t8-get-tcp-listening-connections.sh]: Let's add a separate variable for seconds

**Review by kseniyashaydurova — APPROVED** [review]

File: chapter2/tests/t8/chapter2-l4-t8-get-tcp-listening-connections.sh (hunk 2 above)
Comment (krasninja) [reply chapter2/tests/t8/chapter2-l4-t8-get-tcp-listening-connections.sh]: Fixed.

File: chapter2/tests/t8/chapter2-l4-t8-get-tcp-listening-connections.sh (hunk 3 above)
Comment (krasninja) [reply chapter2/tests/t8/chapter2-l4-t8-get-tcp-listening-connections.sh]: Done.

**Review by dmitry-mightydevops — APPROVED** [review]

**Pull request merged**